docs/user_guide/datasets/acs_income.rst

```diff
@@ -0,0 +1,171 @@
+.. _acsincome_data:
+ACSIncome
```

Comment:
[I haven't gone through the other PR yet so this may be irrelevant]

It seems to me that a user guide item should also provide guidance on when one should (not) use this data set (or at least refer to other resources that cover that).

Reply:
We agree. However, we were thinking that the scope of this PR is just to report the facts of what this dataset is. Any recommendations/interpretations would be a good candidate for a future PR since it's more of an open discussion that definitely requires further analysis to justify any recommendations, especially since UCI Adult (and this updated version) are so central to the fairness literature; UCI Adult appeared in 300+ papers related to fairness according to Ding et al. It's something we're still discussing as a team.

Reply:
That sounds very reasonable. Did you happen to already open an issue for this? That might help to get the discussion started and gather some views from the community.

Reply:
I can open one this afternoon :)